apps/mobile-wallet/src/contexts/walletConnect/WalletConnectContext.tsx

```diff
@@ -211,6 +211,12 @@ export const WalletConnectContextProvider = ({ children }: { children: ReactNode
     }
   }, [dispatch, walletConnectClientInitializationAttempts])
 
+  useEffect(() => {
+    if (walletConnectClientStatus === 'initialized' && !walletConnectClient) {
+      dispatch(walletConnectClientInitializeFailed(t('Lost connection to WalletConnect')))
```

Comment:
This will set the status to `initialization-failed` which will update the UI accordingly and the user won't be stuck at a "success" indication.